.github/workflows/pre-release_trigger.yml

```diff
@@ -0,0 +1,40 @@
+name: Deploy to Staging from Pre-release
+on:
+  release:
+    types: [published]
+
+jobs:
+  setup-release:
+    runs-on: ubuntu-latest
+    outputs:
+      git-tag: ${{ steps.tag.outputs.version }}
+    steps:
+      - name: Init
+        run: |
+          echo Deploy starting
+          # this allows for blame
+          echo ${{ github.actor }}
+      - name: Check release and env values
+        run: |
+          echo "Release tag: ${{ github.event.release.tag_name }}"	          
+          echo "Release name: ${{ github.event.release.name }}"
+          echo "is a prerelease? (github.event.release.prerelease): ${{ github.event.release.prerelease }}"
+          echo "is published (staging) or a release (prod)?: ${{ github.event.action }}"
+
+  pre-release:
+    needs: [setup-release]
+    if: ${{ github.event.release.prerelease == true }}
+    uses: ./.github/workflows/release.yml
+    with:
+      environment: "staging"
+      git-tag: "${{ github.event.release.tag_name }}"
+    secrets: inherit
+
+  release:
+    needs: [setup-release]
+    if: ${{ github.event.release.prerelease == false }}
+    uses: ./.github/workflows/release.yml
+    with:
+      environment: "prod"
+      git-tag: "${{ github.event.release.tag_name }}"
+    secrets: inherit
```

Comment:
why is this still needed here? this workflow should only be for the pre-release.